src/lib/broadcast.jl

```diff
@@ -254,7 +254,14 @@ end
     placeholder = similar(xs)
     sum(xs, dims = dims), Δ -> (placeholder .= Δ,)
   end
-
+  
+  # Make sure sum(f, ::CuArray) uses broadcase through forward-mode defined above
+  # Not the ChainRules.rrule which will use the Zygote.Context and thus not be GPU compatible
+  @adjoint function sum(f, xs::CUDA.CuArray; kws...)
+    @assert !haskey(kws, :init) # TODO add init support (julia 1.6)
+    return pullback(__context__, (f, xs) -> sum(f.(xs); kws...), f, xs)
```

Comment:
You probably don't need the context manager here

Reply:
This is exactly the old code.
I don't really know what it is doing returning the context

Reply:
I think keeping it is better.
It should end up hitting the same code either way, but I can imagine if some specialization are added then the errors we would get are more confusing if we don't include the `__context__`

Reply:
Might be slightly tidier to call `broadcast_forward` directly, instead of going back to pullback. But this is a quick fix to repair an accidental break so the most direct restoration is fine too.